adapter/ipListChecker/adapter.go

```diff
@@ -15,74 +15,93 @@
 package ipListChecker
 
 import (
-	"errors"
+	"fmt"
+	"net/http"
 	"net/url"
 	"time"
 
-	"istio.io/mixer/pkg/adapter"
+	"github.com/golang/protobuf/proto"
+
+	"istio.io/mixer/pkg/aspect/listChecker"
+	"istio.io/mixer/pkg/aspectsupport"
+)
+
+const (
+	// ImplName is the canonical name of this implementation
+	ImplName = "istio/IPListChecker"
 )
 
-// AdapterConfig is used to configure a adapter.
-type AdapterConfig struct {
+// Register registration entry point
+func Register(r aspectsupport.Registry) {
+	r.RegisterCheckList(&adapterState{})
 }
 
 type adapterState struct{}
 
-// NewAdapter returns a Adapter
-func NewAdapter() adapter.Adapter {
-	return &adapterState{}
-}
-
 func (a *adapterState) Name() string {
-	return "IPListChecker"
+	return ImplName
 }
 
 func (a *adapterState) Description() string {
 	return "Checks whether an IP address is present in an IP address list."
 }
 
-func (a *adapterState) DefaultAdapterConfig() adapter.AdapterConfig {
-	return &AdapterConfig{}
+func (a *adapterState) DefaultConfig() proto.Message {
+	return &Config{
+		ProviderUrl:     "http://localhost",
+		RefreshInterval: 60,
+		Ttl:             120,
+	}
 }
 
-func (a *adapterState) ValidateAdapterConfig(config adapter.AdapterConfig) error {
-	_ = config.(*AdapterConfig)
-	return nil
-}
+func (a *adapterState) ValidateConfig(cfg proto.Message) (err error) {
+	c, ok := cfg.(*Config)
+	if !ok {
+		return fmt.Errorf("Invalid message type %#v", cfg)
```

Comment:
The error message should be self contained within a log statement. We should not assume any context when people scanning the logs. Things like "invalid message type xxx" is not meaningful to solve the problem. Something like "Invalid config message type xxx. IPListChecker requires config message type xxx."